noir_stdlib/src/convert.nr

```diff
@@ -0,0 +1,56 @@
+trait From<T> {
+    fn from(input: T) -> Self;
+}
+
+impl<T> From<T> for T {
+  fn from(input: T) -> T {
+      input
+  }
+}
```

Comment:
Can you add a test testing this and a few other impls?

I worry an impl for all `T` would conflict with other impls for all `T`s, although it may not be caught due to the trait being generic. I doubt our coherence check is correct for generic types now that I think about it. I suppose the real concern then is that a similar but actually conflicting impl won't be caught depending on its generics.

Reply:
Nevermind, we can leave the coherence check issue to a separate issue. I'm not sure adding a test here would be necessary. Since its in the stdlib, if any impls did conflict we'd get an error in every test.